Macros/Macros.js

```diff
@@ -192,6 +223,64 @@ class FurnaceMacros {
         div.find(".furnace-macro-expand").on('click', (ev) => div.toggleClass("fullscreen"));
         refreshHighlight();
     }
+
+    static async _execute(message) {
+        let macro = game.macros.getName(message.macroName);
+        if (!macro || !getProperty(macro.data, "flags.furnace.gmexecute" || !game.users.get(macro.data.author).isGM)) {
+            ui.notification.error(`User ${message.userId} tried to invailidly GM execute macro ${message.macroName}`);
+            return "Invalid Request";
+        }
+
+        try {
+            return await macro.renderContent("RemoteContext", message, ...message.args);
+        } catch (err) {
+            ui.notifications.error(`There was an error in your macro syntax. See the console (F12) for details`);
+            console.error(err);
+            return "Macro syntax error";
+        }
+    }
+
+    // request remote execution of a macro
+    static async requestRemoteExecute(macroName, ...args) {
+        let targetGM = game.users.entities.find(u => u.isGM && u.active);
+        let macroMessage = {
+            action: "GMExecuteMacro", 
+            requestId: "remoteResult" + game.user.id + Date.now(), 
+            userId: game.user.id, 
+            targetGM: targetGM._id, 
+            macroName, 
+            sceneId: canvas.scene.id, // pass this so macros can know which scene the player is on.
+            tokenId: ChatMessage.getSpeaker().token, 
+            event: event || {event: {shiftKey: true}}, // this is to keep minor-qol happy
+            args
+        };
+        if (game.user.isGM) return await FurnaceMacros._execute(macroMessage); 
+        // setup to handle exectuion response
+        let requestResponse = await new Promise((resolve, reject) => {
+            Hooks.once(macroMessage.requestId, resolve);
+            game.socket.emit(FurnaceMacros._moduleSocket, macroMessage)
+            setTimeout(() => Hooks.callAll(macroMessage.requestId, "timeout"), 5000);
```

Comment:
ok, this is the 'chewing gum and string' RPC method you created.
I really don't like it. First of all, I thought we could do a `game.socket.emit` and give it a callback for the response as the 3rd argument, but apparently that doesn't work, which I think we may want to bug Atropos about.
The use of `Hooks` is what I don't like the most. The response should be sent back using game.socket.emit, and we should be able to listen to it, and have a map of our pending requests.
One issue is that the hook name `remoteResult-${id}` is too vague, so having it perhaps as `furnaceExecuteMacroResult-${id}` would have made more sense, but in this case, I'll redo this to use a different message action name instead.
I like the use of a timeout. I was wondering how to check for no responses. I'd say that having "timeout" as the response text is not a good idea though. I'd rather it show a `ui.notification` and return an empty string instead, as that could mess up some chat content otherwise.